src/Compilers/CSharp/Portable/Binder/Binder_Crefs.cs

```diff
@@ -838,7 +836,12 @@ private Symbol ConstructWithCrefTypeParameters(int arity, TypeArgumentListSyntax
                 SeparatedSyntaxList<TypeSyntax> typeArgumentSyntaxes = typeArgumentListSyntax.Arguments;
                 var typeArgumentsWithAnnotations = ArrayBuilder<TypeWithAnnotations>.GetInstance(arity);
 
-                DiagnosticBag unusedDiagnostics = DiagnosticBag.GetInstance();
+                var unusedDiagnostics =
+#if DEBUG
+                    new BindingDiagnosticBag(DiagnosticBag.GetInstance());
+#else
+                    BindingDiagnosticBag.Discarded;
```

Comment:
>Discarded [](start = 41, length = 9)

Won't this accumulate diagnostics and never clear them (in non-debug mode)?

Reply:
> Won't this accumulate diagnostics and never clear them (in non-debug mode)?

```BindingDiagnosticBag.Discarded``` doesn't accumulate anything.

---
In reply to: [361552564](https://github.com/dotnet/roslyn/pull/40560#discussion_r361552564) [](ancestors = 361552564)